workbench/src/renderer/components/LogTab/index.jsx

```diff
@@ -74,12 +76,35 @@ export default class LogTab extends React.Component {
     // If we're re-running a model after loading a recent run,
     // we should clear out the logdata when the new run is launched.
     if (this.props.executeClicked && !prevProps.executeClicked) {
-      this.setState({ logdata: '' });
+      this.setState({ logdata: [] });
     }
   }
 
   componentWillUnmount() {
     ipcRenderer.removeAllListeners(`invest-stdout-${this.props.tabID}`);
+    clearTimeout(this.validationTimer);
+  }
+
+  updateState() {
+    this.setState((state) => ({
+      logdata: state.logdata.concat(this.cache)
+    }));
+    this.cache = [];
+  }
+
+  /*
+   * Cache incoming logger messages in real-time, then move them to react state
+   * in batches, at some longer interval. This limits the number of renders in
+   * the event of very high volume logging, while still allowing the browser to
+   * appear that it is updating in real-time.
+   */
+  debouncedLogUpdate(data) {
+    if (this.validationTimer) {
+      clearTimeout(this.validationTimer);
+    }
+    this.cache = [...this.cache, data];

```

Comment:
Isn't this creating a whole new array each time we want to append to it?  If so, is this behavior preferable to simply appending to the existing array?

I see above on line 92 that we're clearing the cache each time we update state from the cache data (totally makes sense), so maybe this really isn't likely to be an issue in practice since the arrays are probably fairly small.

Reply:
Thanks, @phargogh , good catch. My best guess is that I was just in the mode of not mutating data, as that is important for data in react's `state`. But this cache is deliberately not in react state. I changed it to push to the existing array.